mParticle-Apple-SDK/Ecommerce/MPCommerceEvent.mm

```diff
@@ -263,8 +261,8 @@ - (NSMutableArray *)productsList {
 
 #pragma mark Private methods
 - (void)setEventType {
-    static const vector<EventType> productActionEventType {AddToCart, RemoveFromCart, AddToWishlist, RemoveFromWishlist, Checkout, CheckoutOption, Click, ViewDetail, Purchase, Refund};
-    static const vector<EventType> promotionActionEventType {PromotionClick, PromotionView};
+    static const vector<MPEventType> productActionEventType {MPEventTypeAddToCart, MPEventTypeRemoveFromCart, MPEventTypeAddToWishlist, MPEventTypeRemoveFromWishlist, MPEventTypeCheckout, MPEventTypeCheckoutOption, MPEventTypeClick, MPEventTypeViewDetail, MPEventTypePurchase, MPEventTypeRefund};
```

Comment:
We should be able to get rid of this use of std::vector here as well, looks like it's pointless as we just immediately cast back to Obj-C enum type.

Reply:
It still requires a fair bit of casting but at least its all Objective C now.